de.dlr.sc.virsat.graphiti.test/src/de/dlr/sc/virsat/graphiti/diagram/BeanIndependenceSolverTest.java

```diff
@@ -76,91 +89,125 @@ protected void doExecute() {
+		sei2 = ed2.getStructuralElementInstance();
+		sei1.setUuid(new VirSatUuid(UUID));
+		ed1.add(testDocument2);
 
-		caKey = beanIndependenceSolver.getKeyForBusinessObject(testDocument);
-		assertEquals(DOCUMENTUUID, caKey);
+		command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed1.getStructuralElementInstance());
+	}
 
-		Command command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed.getStructuralElementInstance());
-		editingDomain.getCommandStack().execute(command);
+	@Test
+	public void testSolveIndependenceForSei() throws CoreException, IOException {
+		String seiKey = beanIndependenceSolver.getKeyForBusinessObject(ed1);
+		assertEquals(UUID, seiKey);
 
-		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
-			@Override
-			protected void doExecute() {
-				rs.getAndAddStructuralElementInstanceResource(sei2);
-			}
-		});
+		editingDomain.getCommandStack().execute(command);
+		getAndAddStructuralElementInstance(sei2);
 
-		Resource resEd2 = rs.getStructuralElementInstanceResource(ed2.getStructuralElementInstance());
-		resEd2.getContents().add(ed2.getStructuralElementInstance());
+		addStructuralElementToResource(ed2);
 		Object sei2Object = beanIndependenceSolver.getBusinessObjectForKey(ed2.getUuid());
 		ABeanStructuralElementInstance aBean = (ABeanStructuralElementInstance) sei2Object;
 		assertEquals(ed2, aBean);
-		
-		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
-			@Override
-			protected void doExecute() {
-				rs.getAndAddStructuralElementInstanceResource(sei);
-			}
-		});
 
-		Boolean permission = DiagramHelper.hasDiagramWritePermission(sei);
-		assertEquals(true, permission);
-		
-		Resource resEd = rs.getStructuralElementInstanceResource(ed.getStructuralElementInstance());
-		resEd.getContents().add(ed.getStructuralElementInstance());
+		getAndAddStructuralElementInstance(sei1);
+		checkDiagramWritePermissionForStructuralElement(sei1);
+
+		addStructuralElementToResource(ed1);
 		Object obj = beanIndependenceSolver.getBusinessObjectForKey(seiKey);
 		aBean = (ABeanStructuralElementInstance) obj;
-		assertEquals(ed, aBean);
+		assertEquals(ed1, aBean);
+	}
+
+	@Test
+	public void testSolveIndependenceForBeanProperty() throws CoreException, IOException {
+		editingDomain.getCommandStack().execute(command);
+		getAndAddStructuralElementInstance(sei2);
+		addStructuralElementToResource(ed2);
+		getAndAddStructuralElementInstance(sei1);
+		checkDiagramWritePermissionForStructuralElement(sei1);
+		addStructuralElementToResource(ed1);
 
+		//test property
 		Object documentObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getUuid());
 		ABeanCategoryAssignment aCategory = (ABeanCategoryAssignment) documentObj;
 		assertEquals(testDocument2, aCategory);
-		
+
+		Object beanNoteObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getNoteBean().getUuid());

```

Comment:
I think testing only one property bean is enough.
We just need to test here that the bean independence solver actually handles properties.
That the individual types are handled correctly is tested in the bean factory.